grid2op/Reward/AlertReward.py

```diff
@@ -0,0 +1,263 @@
+        return first_attack_step_in_window
+
+    def _get_nb_of_alerts_matching_an_attack(self): 
+        attacked_lines_first_step =  self._fist_attack_step_in_time_window[self._fist_attack_step_in_time_window>=0]
+        alert_on_attacked_lines =  self._line_alerts_in_time_window[self._fist_attack_step_in_time_window>=0]
+        
+        attacked_lines_with_alert = np.array([alert_on_attacked_lines[line, step] for line, step in enumerate(attacked_lines_first_step)])
+        nb_correct_alert = attacked_lines_with_alert.sum()
+        return nb_correct_alert
+
+
+    def __call__(self, action, env, has_error, is_done, is_illegal, is_ambiguous):
+        legal_alert_action = env._attention_budget._last_alert_action_filtered_by_budget
+
+        if env.nb_time_step == self.current_step_first_encountered+1:
+            self.current_step_first_encountered = env.current_obs.current_step
+            self._step_update(legal_alert_action, action, env)
+
+        score = self.reward_min
+
+        if  is_done & (not has_error): 
+            # end of episode 
+            score = self.reward_end_episode_bonus
+        elif self.is_in_blackout(has_error, is_done): 
+            # If blackout
+            # Is the blackout caused by an attack ? 
+            # Has there been an attack in the last ``ALERT_TIME_WINDOW`` time steps ? 
+            if self._has_attack_in_time_window.any() :
+                
+                nb_correct_alerts = self._get_nb_of_alerts_matching_an_attack()
+                score = self.reward_min_blackout + self.reward_unit_step * nb_correct_alerts
+                # If we correctly predict 2 out of 3 we get - 2
+                # If we correctly predict  1 out of 3 we get - 6
+                # If we correctly predict  0 out of 3 we get - 10 
+
+        else: 
+            # If there is no blackout 
+            if self._has_attack_at_first_time_in_window.any() & self._is_first_step_of_attack.any():
+                # If there is an attack
+
+                alerts_with_attack_at_first_time_in_window =self._line_alerts_in_time_window[:,0][self._has_attack_at_first_time_in_window]
+                # As there is no blackout, we do not want to raise any alert 
+                if alerts_with_attack_at_first_time_in_window.any():
+                    nb_alerts_for_attacks = alerts_with_attack_at_first_time_in_window.sum()
+                    # If there is an alert raised for one of the attacked line, we give the minimal reward
+                    nb_of_attacks = self._has_attack_at_first_time_in_window.sum()
+                    score = self.reward_max_no_blackout - (nb_alerts_for_attacks/nb_of_attacks) * (self.reward_max_no_blackout  - self.reward_min_no_blackout)
+                else : 
+                    # If we don't raise any alert, we are happy with it, so we give the maximal reward value                    
+                    score = self.reward_max_no_blackout
+ 
+        return score
```

Comment:
score should be 0 if `if self._has_attack_at_first_time_in_window.any() & self._is_first_step_of_attack.any():` is `False` but from my understanding its ` score = self.reward_min`.

And reward_min should not be 0. but the min between the 4 parameters (same for the reward max by the way)